models.py

```diff
@@ -0,0 +1,159 @@
+        return mw_dict
+
+    @classmethod
+    def from_dict(cls, mw_dict, controller):
+        """Create a maintenance window from a dictionary of attributes."""
+        mw_id = mw_dict.get('id')
+
+        start = datetime.datetime.strptime(mw_dict['start'], TIME_FMT).astimezone(pytz.utc)
+        end = datetime.datetime.strptime(mw_dict['end'], TIME_FMT).astimezone(pytz.utc)
+        items = list()
+        for i in mw_dict['items']:
+            try:
+                item = cls.uni_from_dict(i, controller)
+            except KeyError:
+                item = cls.link_from_dict(i, controller)
+            except TypeError:
+                item = i
+            if item is None:
+                return None
+            items.append(item)
+        return cls(start, end, controller, items, mw_id)
+
+    def update(self, mw_dict, controller):
+        """Update a maintenance window with the data from a dictionary."""
+        if 'start' in mw_dict:
+            self.start = datetime.datetime.strptime(mw_dict['start'], TIME_FMT).astimezone(pytz.utc)
+        if 'end' in mw_dict:
+            self.end = datetime.datetime.strptime(mw_dict['end'], TIME_FMT).astimezone(pytz.utc)
+        if 'items' in mw_dict:
+            items = list()
+            for i in mw_dict['items']:
+                try:
+                    item = self.uni_from_dict(i, controller)
+                except KeyError:
+                    item = self.link_from_dict(i, controller)
+                except TypeError:
+                    item = i
+                if item:
+                    items.append(item)
+            self.items = items
+
+    @staticmethod
+    def intf_from_dict(intf_id, controller):
+        """Get the Interface instance with intf_id."""
+        intf = controller.get_interface_by_id(intf_id)
+        return intf
+
+    @staticmethod
+    def uni_from_dict(uni_dict, controller):
+        """Create UNI instance from a dictionary."""
+        intf = MaintenanceWindow.intf_from_dict(uni_dict['interface_id'],
+                                                controller)
+        tag = TAG.from_dict(uni_dict['tag'])
+        if intf and tag:
+            return UNI(intf, tag)
+        return None
+
+    @staticmethod
+    def link_from_dict(link_dict, controller):
+        """Create a link instance from a dictionary."""
+        endpoint_a = controller.get_interface_by_id(
+            link_dict['endpoint_a']['id'])
+        endpoint_b = controller.get_interface_by_id(
+            link_dict['endpoint_b']['id'])
+
+        link = Link(endpoint_a, endpoint_b)
+        if 'metadata' in link_dict:
+            link.extend_metadata(link_dict['metadata'])
+        s_vlan = link.get_metadata('s_vlan')
+        if s_vlan:
+            tag = TAG.from_dict(s_vlan)
+            link.update_metadata('s_vlan', tag)
+        return link
+
+    def start_mw(self):
+        """Actions taken when a maintenance window starts."""
+        pass
```

Comment:
I guess the linter will complain about 'unnecessary-pass'.

Reply:
This is something that will be implemented later, but I needed the function defined.